File: backend/backend-node/src/s3.mjs

import config from "./config.mjs";
import { S3Client, PutObjectCommand, DeleteObjectCommand } from "@aws-sdk/client-s3";

//recibe un buffer y lo sube a S3
const uploadImageS3 = async (buff, path) => {
    const client = new S3Client({
      region: config.region,
      credentials: {
        accessKeyId: config.accessKeyId,
        secretAccessKey: config.secretAccessKey,
      },
    });
  
    const command = new PutObjectCommand({
      Bucket: config.bucket,
      Key: path,
      Body: buff,
      ContentType: "image/jpeg",
    });
  
    try {
      const response = await client.send(command);
      return response;
    } catch (error) {
      console.error(error);
      return null;
    }
  };

const uploadMP3S3 = async (buff, path) => {
    const client = new S3Client({
      region: config.region,
      credentials: {
        accessKeyId: config.accessKeyId,
        secretAccessKey: config.secretAccessKey,
      },
    });
  
    const command = new PutObjectCommand({
      Bucket: config.bucket,
      Key: path,
      Body: buff,
      ContentType: "audio/mpeg",
    });
  
    try {
      const response = await client.send(command);
      return response;
    } catch (error) {
      console.error(error);
      return null;
    }
  };

  const deleteObjectS3 = async (path) => {
    //url = "https://multimedia-semi1-seccion-a-g11.s3.us-east-1.amazonaws.com/Fotos/Vermastarde320240830182758.jpg"
    let parts = path.split('/');
    parts = parts[parts.length - 2] + "/" + parts[parts.length - 1];
  
    const client = new S3Client({
      region: config.region,
      credentials: {
        accessKeyId: config.accessKeyId,
        secretAccessKey: config.secretAccessKey,
      },
    });
    const command = new DeleteObjectCommand({
      Bucket: config.bucket,
      Key: parts,
    });
  
    try {
      const response = await client.send(command);
      return response;
    } catch (error) {
      console.error(error);
      return null;
    }
  };
  
  export { uploadImageS3, uploadMP3S3,  deleteObjectS3};
  

File: backend/backend-python/src/s3.py
import boto3
from botocore.exceptions import NoCredentialsError, PartialCredentialsError
from config import config

def uploadImageS3(buff, path):
    # Configuración del cliente S3
    s3_client = boto3.client(
        's3',
        region_name = config["region"], 
        aws_access_key_id = config["accessKeyId"],
        aws_secret_access_key = config["secretAccessKey"]
    )

    try:
        # Subir la imagen a S3
        response = s3_client.put_object(
            Bucket= config["bucket"],
            Key=path,
            Body=buff,
            ContentType='image/jpeg'
        )
        return response
    except (NoCredentialsError, PartialCredentialsError) as e:
        print("Error de credenciales:", e)
        return None
    except Exception as e:
        print("Error al subir la imagen:", e)
        return None
  
def uploadMP3S3(buff, path):
    # Configuración del cliente S3
    s3_client = boto3.client(
        's3',
        region_name = config["region"], 
        aws_access_key_id = config["accessKeyId"],
        aws_secret_access_key = config["secretAccessKey"]
    )

    try:
        # Subir la imagen a S3
        response = s3_client.put_object(
            Bucket= config["bucket"],
            Key=path,
            Body=buff,
            ContentType='audio/mpeg'
        )
        return response
    except (NoCredentialsError, PartialCredentialsError) as e:
        print("Error de credenciales:", e)
        return None
    except Exception as e:
        print("Error al subir la imagen:", e)
        return None

def deleteObjectS3(path):

    parts = path.split('/')
    parts = parts[-2] + "/" + parts[-1]

    # Configuración del cliente S3
    s3_client = boto3.client(
        's3',
        region_name = config["region"], 
        aws_access_key_id = config["accessKeyId"],
        aws_secret_access_key = config["secretAccessKey"]
    )

    try:
        # Eliminar el objeto de S3
        response = s3_client.delete_object(
            Bucket= config["bucket"],
            Key=parts
        )
        return response
    except (NoCredentialsError, PartialCredentialsError) as e:
        print("Error de credenciales:", e)
        return None
    except Exception as e:
        print("Error al eliminar objeto:", e)
        return None